includes/widgets/class-wc-widget-price-filter.php

```diff
@@ -68,23 +68,45 @@ public function widget( $args, $instance ) {
 			return;
 		}
 
-		$min_price = isset( $_GET['min_price'] ) ? wc_clean( wp_unslash( $_GET['min_price'] ) ) : null; // WPCS: input var ok, CSRF ok.
-		$max_price = isset( $_GET['max_price'] ) ? wc_clean( wp_unslash( $_GET['max_price'] ) ) : null; // WPCS: input var ok, CSRF ok.
-		if ( ! wc()->query->get_main_query()->post_count && null === $min_price && null === $max_price ) {
+		// If there are not posts and we're not filtering, hide the widget.
+		if ( ! WC()->query->get_main_query()->post_count && ! isset( $_GET['min_price'] ) && ! isset( $_GET['max_price'] ) ) { // WPCS: input var ok, CSRF ok.
 			return;
 		}
 
 		wp_enqueue_script( 'wc-price-slider' );
 
+		// Round values to nearest 10 by default.
+		$step = max( apply_filters( 'woocommerce_price_filter_widget_step', 10 ), 1 );
+
 		// Find min and max price in current result set.
-		$prices = $this->get_filtered_price();
-		$min    = floor( $prices->min_price );
-		$max    = ceil( $prices->max_price );
+		$prices    = $this->get_filtered_price();
+		$min_price = $prices->min_price;
+		$max_price = $prices->max_price;
+
+		// Check to see if we should add taxes to the prices if store are excl tax but display incl.
+		$tax_display_mode = get_option( 'woocommerce_tax_display_shop' );
+
+		if ( wc_tax_enabled() && ! wc_prices_include_tax() && 'incl' === $tax_display_mode ) {
+			$tax_class = apply_filters( 'woocommerce_price_filter_widget_tax_class', '' ); // Uses standard tax class.
+			$tax_rates = WC_Tax::get_rates( $tax_class );
 
-		if ( $min === $max ) {
+			if ( $tax_rates ) {
+				$min_price += WC_Tax::get_tax_total( WC_Tax::calc_exclusive_tax( $min_price, $tax_rates ) );
```

Comment:
Similarly to adjusting max in `get_tax_query`, maybe we should not adjust min here in case we have low priced products with zero/reduced tax rate?

Reply:
See comment above.